pkg/pquerier/select.go

```diff
@@ -408,27 +433,31 @@ type partQuery struct {
 	attrs      []string
 	step       int64
 
-	chunk0Time        int64
-	chunkTime         int64
-	preAggregated     bool
-	aggregationParams *aggregate.AggregationParams
+	chunk0Time              int64
+	chunkTime               int64
+	useServerSideAggregates bool
+	aggregationParams       *aggregate.AggregationParams
 }
 
-func (query *partQuery) getItems(ctx *selectQueryContext, name string, aggregatesAndChunk bool) error {
+func (query *partQuery) getItems(ctx *selectQueryContext, name string, groupBy []string, aggregatesAndChunk bool) error {
 
 	path := query.partition.GetTablePath()
+	if query.useServerSideAggregates && !aggregatesAndChunk && len(groupBy) > 0 {
```

Comment:
a bit petty but, i would move the `len(groupBy) > 0` part to be the first condition in the `if` because first of all, this is the real essence of the condition. Plus because we're such a high performance product, and mostly this condition would "fail" the if it should be first 😅

Reply:
Better yet - I'll avoid resolving the groupBy labels altogether in that case.